staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -622,22 +624,26 @@ func (r *crdHandler) getOrCreateServingInfoFor(uid types.UID, name string) (*crd
 
 	structuralSchemas := map[string]*structuralschema.Structural{}
 	for _, v := range crd.Spec.Versions {
-		val, err := apiextensions.GetSchemaForVersion(crd, v.Name)
+		val, err := apiextensionshelpers.GetSchemaForVersion(crd, v.Name)
 		if err != nil {
 			utilruntime.HandleError(err)
 			return nil, fmt.Errorf("the server could not properly serve the CR schema")
 		}
 		if val == nil {
 			continue
 		}
-		s, err := structuralschema.NewStructural(val.OpenAPIV3Schema)
-		if *crd.Spec.PreserveUnknownFields == false && err != nil {
+		internalValidation := &apiextensionsinternal.CustomResourceValidation{}
+		if err := apiextensions.Convert_v1_CustomResourceValidation_To_apiextensions_CustomResourceValidation(val, internalValidation, nil); err != nil {
```

Comment:
highlighting the explicit conversion v1->internal. we are validating structral-schema and openapi-schema on internal types, which is not changed in this PR

Reply:
@sttts PTAL

Reply:
wouldn't mind to switch that in a follow-up as well. But Sgtm.